security/integration/main.go

```diff
@@ -143,77 +150,138 @@ func initializeIntegrationTest(cmd *cobra.Command, args []string) error {
 
 func runSelfSignedCATests() error {
 	// Create Istio CA with self signed certificates
-	self_signed_ca_pod, err := utils.CreatePod(opts.clientset, opts.namespace,
+	selfSignedCaPod, err := utils.CreatePod(opts.clientset, opts.namespace,
 		fmt.Sprintf("%v/istio-ca:%v", opts.containerHub, opts.containerTag), "istio-ca-self")
 	if err != nil {
 		return fmt.Errorf("failed to deploy Istio CA (error: %v)", err)
 	}
-	glog.Infof("Succesfully created Istio CA pod(%v) with the self-signed-ca option",
-		self_signed_ca_pod.GetName())
+	glog.Infof("Successfully created Istio CA pod(%v) with the self-signed-ca option",
+		selfSignedCaPod.GetName())
 
 	// Test the existence of istio.default secret.
-	if s, err := utils.WaitForSecretExist(opts.clientset, opts.namespace, "istio.default",
-		secretWaitTime); err != nil {
+	s, err := utils.WaitForSecretExist(opts.clientset, opts.namespace, "istio.default",
+		secretWaitTime)
+	if err != nil {
 		return err
-	} else {
-		glog.Info(`Secret "istio.default" is correctly created`)
+	}
 
-		expectedID := fmt.Sprintf("spiffe://cluster.local/ns/%s/sa/default", s.GetNamespace())
-		examineSecret(s, expectedID)
+	glog.Info(`Secret "istio.default" is correctly created`)
+	_, _, _, err = examineSecret(s)
+	if err != nil {
+		return err
 	}
 
 	// Delete the secret.
 	do := &metav1.DeleteOptions{}
 	if err := opts.clientset.CoreV1().Secrets(opts.namespace).Delete("istio.default", do); err != nil {
 		return err
-	} else {
-		glog.Info(`Secret "istio.default" has been deleted`)
 	}
+	glog.Info(`Secret "istio.default" has been deleted`)
 
 	// Test that the deleted secret is re-created properly.
 	if _, err := utils.WaitForSecretExist(opts.clientset, opts.namespace, "istio.default",
 		secretWaitTime); err != nil {
 		return err
-	} else {
-		glog.Info(`Secret "istio.default" is correctly re-created`)
 	}
+	glog.Info(`Secret "istio.default" is correctly re-created`)
+
+	return nil
+}
 
-	// Delete pods
-	err = utils.DeletePod(opts.clientset, opts.namespace, self_signed_ca_pod.GetName())
+func cleanUpSelfSignedCATests() {
+	glog.Info("Removing pods: istio-ca-self")
+	_ = utils.DeletePod(opts.clientset, opts.namespace, "istio-ca-self")
+	glog.Info("Removing secrets: istio.default")
+	_ = utils.DeleteSecrets(opts.clientset, opts.namespace, "istio.default")
```

Comment:
Could you also remove istio-ca-cert? It's used to store the CA cert/key.

Reply:
Done